### ISSUE TITLE
Consider adding Backstack.services() method as in GlobalServices.services()

### ISSUE THREAD

**Issue by ElianFabian** [Zhuinden/simple-stack#284]
I think it would be helpful to have a way of getting a list of all the currently available services with the Backstack, similar to GlobalServices.services().

A use case for this could be implementing custom callbacks for services, similar to ScopedServices.Registered.

For example, let's say I want a service to be notified when Activity.onNewIntent() is called, It would look like this:

```kt
override fun onNewIntent(intent: Intent) {
    super.onNewIntent(intent)

    backstack.services().forEach { (key, service) ->
       if (service is OnNewIntentListener) {
           service.onNewIntent(intent)
       }
    }
}
```

I'm not sure if there would be a better way to achieve a similar result with ease.

**Comment by Zhuinden** [comment]
Haven't really had time to work on this but I hope to get it done by the end of this week at the latest.

**Comment by Zhuinden** [comment]
Released in 2.9.0.

**Comment by Zhuinden** [comment]
https://github.com/Zhuinden/simple-stack/issues/277 vaguely also applies.
